src/persist-client/src/internal/encoding.rs

```diff
@@ -810,12 +906,50 @@ impl<T: Timestamp + Lattice + Codec64> RustType<ProtoStateRollup> for UntypedSta
             hostname: x.hostname,
             collections,
         };
-        Ok(UntypedState {
-            state,
-            key_codec: x.key_codec.into_rust()?,
-            val_codec: x.val_codec.into_rust()?,
-            ts_codec: x.ts_codec.into_rust()?,
-            diff_codec: x.diff_codec.into_rust()?,
+
+        let diffs: Option<InlinedDiffs> = x.diffs.map(|diffs| diffs.into_rust()).transpose()?;
+        if let Some(diffs) = &diffs {
+            if diffs.lower != state.latest_rollup().0.next() {
+                return Err(TryFromProtoError::InvalidPersistState(format!(
+                    "diffs lower ({}) should match latest rollup's successor: ({})",
+                    diffs.lower,
+                    state.latest_rollup().0.next()
+                )));
```

Comment:
Can we test this error case explicitly?

Reply:
I think I'm torn on this one -- to verify it, we'd have to write a novel serialization path that intentionally persists the wrong data. It feels a bit like a test that would be verifying itself as much as anything?